rules/mock_wrap_watcher.go

```diff
@@ -17,10 +18,12 @@ func (ww *MockWatcherWrapper) Watch(ctx context.Context, key string, opts ...cli
 	watcherChan := ww.KvWatcher.Watch(ctx, key, opts...)
 	ww.Logger.Info("initiating watch", zap.String("key", key))
 	go func() {
-		watcherResp := <-watcherChan
-		ww.Logger.Info("watch response received", zap.String("resp", key))
-		ww.Responses = append(ww.Responses, watcherResp)
-		c <- watcherResp
+		for {
```

Comment:
The original code only forwarded the first watch event and the test using this was relying on the crawler to finish the test.